.github/workflows/obs-upload.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+set -e
+
+RELEASES='rhel:7 rhel:8 fedora:32 fedora:33'
+
+for i in git curl xmlstarlet rpmspec
+do
+    if test -z "$(which "$i" || true)"
+    then
+        echo "The $i binary could not be found. Aborting."
+        exit 1
+    fi
+done
+
+BASE="$(git rev-parse --show-toplevel 2> /dev/null)"
+SPECFILE="${BASE}/ungoogled-chromium.spec"
+
+if test -z "${BASE}"
+then
+    echo "BASE directory could not be determined. Aborting."
+    exit 1
+fi
+
+if test ! -f "${SPECFILE}"
+then
+    echo "${SPECFILE} must exist and be a regular file. Aborting."
+    exit 1
+fi
+
+for i in OBS_API_USERNAME OBS_API_PASSWORD
+do
+    if test -z "$(eval echo \$${i})"
+    then
+        echo "$i is not in the environment. Aborting."
+        exit 1
+    fi
+done
+
+PROJECT="${OBS_API_PROJECT:-home:${OBS_API_USERNAME}}"
+
+curl()
+{
+    for i in `seq 1 5`
+    do
+        command curl -sS -K - "${@}" << EOF
+user="${OBS_API_USERNAME}:${OBS_API_PASSWORD}"
+EOF
+        test "${?}" -eq 0 && return 0
+        sleep 30s
+    done
+    return 1
+}
+
+get_type()
+{
+    local TAG
+    local TYPE
+
+    TAG="$(git describe --tags --exact-match 2> /dev/null || true)"
+
+    if test -z "${TAG}"
+    then
+        TYPE='development'
+    else
+        TYPE='production'
+    fi
+
+    echo "${TYPE}"
+}
+
+generate_obs()
+{
+    local ROOT="${1}"
+    local release
+    local version
+    local filename
+    local url
+    local protocol
+    local host
+    local path
+    local i
+
+    sed -r '1c%global obs 1' "${SPECFILE}" > "${ROOT}/ungoogled-chromium.spec"
+    SPECFILE="${ROOT}/ungoogled-chromium.spec"
+
+    echo -n > urls
+
+    for i in ${RELEASES}
```

Comment:
Why there is a for loop for `${RELEASES}`? Are there multiple releases building at the same time?

Reply:
In order to get the complete list of source files to build into the _service file I have to parse it by pretending to be each of the releases that we support. This is mainly because some source files are hidden behind release specific macros.